mlflow/tensorflow/__init__.py

```diff
@@ -782,6 +797,39 @@ def train(original, self, *args, **kwargs):
 
         result = original(self, *args, **kwargs)
 
+        def _extract_data_from_input_fn():
+            """
+            Extracts sample data from dict (str -> ndarray) or Tensor type.
+
+            :return: a slice (of limit ``mlflow.utils.autologging_utils.INPUT_EXAMPLE_SAMPLE_ROWS``)
+                     of the input of type `np.ndarray`
+            """
+            from mlflow.tensorflow._autolog import (
+                extract_sample_dict_or_tensor,
+                extract_sample_features_from_batched_tf_dataset,
+            )
+
+            input_training_data = kwargs.get("input_fn")()
+            input_features = None
+            if isinstance(input_training_data, tuple):
+                input_features = extract_sample_dict_or_tensor(input_training_data[0])
+            elif isinstance(input_training_data, tensorflow.data.Dataset):
+                input_features = extract_sample_features_from_batched_tf_dataset(
+                    input_training_data
+                )
+            else:
+                warnings.warn(
+                    "Tensorflow autologging doesn't support this "
+                    f"type for input example: {type(input_training_data)}."
+                )
+            return input_features
```

Comment:
Can we move this function into `_autolog.py` and have it accept training data input as an argument? Then, can we make the methods it relies on private by prefixing with `_`?  E.g. `extract_sample_dict_or_tensor` -> `_extract_sample_dict_or_tensor`.